chuck-mqtt/src/main/java/com/gojek/chuckmqtt/internal/presentation/transactiondetail/ui/screen/TransactionDetailScreen.kt

```diff
@@ -1,86 +1,137 @@
 import com.gojek.chuckmqtt.internal.presentation.transactiondetail.mvi.TransactionDetailViewState
 import com.gojek.chuckmqtt.internal.presentation.transactiondetail.viewmodel.TransactionDetailViewModel
+import com.gojek.chuckmqtt.internal.utils.indexesOf
 
 @Composable
 internal fun TransactionDetailScreen(
     transactionId: Long,
-    intentLambda: (TransactionDetailIntent) -> Unit,
     transactionDetailViewModel: TransactionDetailViewModel
 ) {
     val state by transactionDetailViewModel.states()
         .subscribeAsState(TransactionDetailViewState.default())
 
+    var queriedPacketBody by remember {
+        mutableStateOf("")
+    }
+
+    transactionDetailViewModel.dispatchIntent(
+        GetTransactionDetailIntent(
+            transactionId
+        )
+    )
+
     ChuckMqttTheme {
         Scaffold(
             topBar = {
                 DetailAppBar(
                     toolBarTitle = state.transaction.packetName,
                     isSent = state.transaction.isSent,
                     onShareClick = {
-                        intentLambda(ShareTransactionDetailIntent(transactionId))
+                        transactionDetailViewModel.dispatchIntent(
+                            ShareTransactionDetailIntent(
+                                transactionId
+                            )
+                        )
+                    },
+                    onQueryTextChange = {
+                        queriedPacketBody = it
                     },
-                    onSearchClick = {
+                    onClearSearchQueryClicked = {
+                        queriedPacketBody = ""
                     }
                 )
             },
             content = {
-                TransactionDetail(transaction = state.transaction)
-
-                LaunchedEffect(Unit) {
-                    intentLambda(
-                        GetTransactionDetailIntent(
-                            transactionId
-                        )
-                    )
-                }
+                TransactionDetail(
+                    transaction = state.transaction,
+                    queriedPackedBody = queriedPacketBody
+                )
             }
         )
     }
 }
 
 @Composable
-private fun TransactionDetail(transaction: MqttTransactionUiModel) {
+private fun TransactionDetail(transaction: MqttTransactionUiModel, queriedPackedBody: String) {
     val packetInfo = HtmlCompat.fromHtml(
         transaction.packetInfo,
         HtmlCompat.FROM_HTML_MODE_LEGACY
     )
     Column(modifier = Modifier.padding(16.dp)) {
         if (packetInfo.isNotEmpty()) {
-            Text(
-                text = packetInfo.toString(),
-                style = TextStyle(fontSize = 14.sp, fontWeight = FontWeight.Normal)
-            )
+            AndroidView(factory = { context -> TextView(context) }, update = {
+                it.text = packetInfo
+            })
```

Comment:
@rizkyfadillah what is the reason behind using AndroidView?

Reply:
Not sure why this was used this before. I'll get back to you mas.